### PR TITLE
fix(forms): language prop can't be a function

### PR DESCRIPTION
**Pull request by lmaillet**
**What is the problem this PR is trying to solve?**
`props.language` is tested to be a function.
Can't use a function in `Object.assign`.

**What is the chosen solution to this problem?**
Remove the `typeof function` test.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by acateland** [comment]
@lmaillet didn't check the rest of the code but maybe the `language` props was indeed intended to be a function or an object and be assigned to the language variable ?

**Comment by lmaillet** [comment]
@acateland 
You mean that language could be a function **OR** an object ?
Something like that:
```
if (typeof props.language === 'function') {
	Object.assign(language, props.language());
} else if (props.language !== undefined) {
	Object.assign(language, ...props.language);
}
```

File: packages/forms/src/UIForm/UIForm.component.js [hunk 1]
```diff
@@ -36,8 +36,8 @@ export class UIFormComponent extends React.Component {
 		this.onActionClick = this.onActionClick.bind(this);
 		// control the tv4 language here.
 		const language = getLanguage(props.t);
-		if (typeof props.language === 'function') {
-			Object.assign(language, props.language);
+		if (props.language !== undefined) {
```
Comment (ncomont) [inline packages/forms/src/UIForm/UIForm.component.js]: you should use `!= null` to check both `undefined`and `null`values.

File: packages/forms/src/UIForm/UIForm.component.js [hunk 2]
```diff
@@ -36,8 +36,8 @@ export class UIFormComponent extends React.Component {
 		this.onActionClick = this.onActionClick.bind(this);
 		// control the tv4 language here.
 		const language = getLanguage(props.t);
-		if (typeof props.language === 'function') {
-			Object.assign(language, props.language);
+		if (props.language != null) {
+			Object.assign(language, ...props.language);
```
Comment (ncomont) [inline packages/forms/src/UIForm/UIForm.component.js]: Just to be sure... Are you sure you really want to spread `props.language`, here ?

Considering : 

```
const a = {aa: 11, bb: 22}
const b = {cc: 33, bb: 99}
```

Then : 

```
Object.assign({}, a, b)
{aa: 11, bb: 99, cc: 33}
```

Whereas :

```
Object.assign({}, a, ...b)
VM9957:1 Uncaught TypeError: undefined is not a function
```

Comment (ncomont) [reply packages/forms/src/UIForm/UIForm.component.js]: (if I'm right, I think that a test is missing 😄)

**Pull request merged**